packages/postgres-database/src/simcore_postgres_database/utils_projects_nodes.py

```diff
@@ -28,6 +28,10 @@ class ProjectNodesNodeNotFound(BaseProjectNodesError):
     ...
 
 
+class ProjectNodesNonUniqueNodeFoundError(BaseProjectNodesError):
```

Comment:
NOTE: If you want to maintain the flexibility for future sharing and backward compatibility by having the same node_id across multiple projects, you only need to add a project_node_id column to the project_metadata. Then, the foreign relationship can be established between project_id and project_node_id (or just the project_node_id, as this already implies that it is unique across projects).

Reply:
not sure I get your point. actually I don't understand the need of that additional column

Reply:
actually the uniqueness is defined by project UUID AND node ID. the node ID is not unique per se.

Reply:
To be precise, I think node_id should be replaced with project_node_id, as this would create a more natural relationship. But let's discuss this in person.

Reply:
discussed :+1: